### PR TITLE
`Programming exercises`: Change the testwise coverage editor to Monaco

### PR DESCRIPTION
**Pull request by pzdr7**
# Stacked on PR #9173 

#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added multiple screenshots/screencasts of my UI changes.

#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server configured with **Gitlab** and **Jenkins**.

### Motivation and Context

The largely unused testwise coverage feature still relies on Ace to display files. Since we want to remove Ace from Artemis, we have to update this component.

### Description

- Changed Ace to Monaco in the testwise coverage view
- Removed a method call from the template

### Steps for Testing

> [!CAUTION]
> The testwise coverage feature is unavailable on the regular test servers. You will have to test this feature on the **Legacy TS2**.

Prerequisites:
- Legacy TS2 (deployment via Bamboo)
- 1 Instructor
- 1 Programming exercise with testwise coverage enabled
  - Note that the feature hasn't been 100% reliable. For convenience, you can use this programming exercise: https://artemis-test2.artemis.in.tum.de/course-management/8641/programming-exercises/13951 (sign in with e.g. test user 11)

1. Log in to Artemis
2. Navigate to course management > Your course > Exercises > Your programming exercise and click on the title of your exercise.
3. Click "Show testwise coverage"
4. Look at a few files and verify that the covered lines are marked. (Note that I did not change the logic that determines which lines are highlighted. See "Screenshots" below for the lines that were originally highlighted)
5. Select/Deselect a few tests. Verify that the covered lines change accordingly in the editors.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| testwise-coverage-file.component.ts | 94.59% | ✅ |
### Screenshots

<details>
  <summary>Original UI (incl. files with covered lines for comparison purposes)</summary>

image

image

image

image

</details>

<details>
  <summary>New UI</summary>

image

</details>

### CODE REVIEW

File: src/main/webapp/app/exercises/programming/hestia/testwise-coverage-report/testwise-coverage-file.component.ts [hunk 1]
```diff
@@ -48,6 +52,7 @@ export class TestwiseCoverageFileComponent implements OnInit, OnChanges {
 
         // set the covered line ratio accordingly
         this.proportionCoveredLines = orderedLines.length / this.fileReport!.lineCount!;
+        this.proportionString = (this.proportionCoveredLines * 100).toFixed(1) + ' %';
```
Comment (pzdr7) [inline src/main/webapp/app/exercises/programming/hestia/testwise-coverage-report/testwise-coverage-file.component.ts:55]:
```suggestion
        this.proportionString = `${(this.proportionCoveredLines * 100).toFixed(1)} %`;
```

File: src/main/webapp/app/exercises/programming/hestia/testwise-coverage-report/testwise-coverage-file.component.scss [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 .mat-expansion-panel .mat-expansion-panel-body {
     padding: 0 !important;
-    text-align: center;
+    text-align: unset;
```
Comment (pzdr7) [inline src/main/webapp/app/exercises/programming/hestia/testwise-coverage-report/testwise-coverage-file.component.scss:7]:
```suggestion
```

**Review by JohannesWt — APPROVED** [review]
Code LGTM 👍

**Review by Jan-Thurner — APPROVED** [review]
tested on Legacy TS2.

Covered lines match screenshots: ✅ 
Selecting and deselecting tests updates covered lines accordingly: ✅

**Review by SimonEntholzer — APPROVED** [review]
Generating the coverage for new exercises, and viewing it, works on the legacy TS2 :+1:

**Review by b-fein — APPROVED** [review]
Code :+1:

**Review by sarpsahinalp — APPROVED** [review]
Code lgtm

**Pull request merged**